src/utils/useAbi.ts

```diff
@@ -0,0 +1,42 @@
+import { JsonFragment } from '@ethersproject/abi'
+import { Fragment } from 'ethers/lib/utils'
+import { useState } from 'react'
+
+export type ABI = ReadonlyArray<Fragment | JsonFragment | string>
+
+export const useAbi = (actorAddr: string) => {
```

Comment:
theres an annoying bug where - when you set your contract in local storage, you have to refresh the page in order for it to take effect. this is because the upload input renders its own `useAbi` hook, and doesn't share the same state with its sibling components that also render the `useAbi` hook 

there are two options i see:

1. make a context wrapper so all useAbi hooks share the same state
2. change the prop structure as to only use the useAbi hook once

I'm thinking 1 probably makes more sense?